### ISSUE TITLE
Can provide arguments for closed records

### ISSUE THREAD

**Issue by mgold** [elm-lang/elm-compiler#667]
````
type Y = { y : Int }
type Y2 = Y {}
type Y3 = Y {} {}
````
No complaints from the compiler. Unless I'm missing something, this shouldn't be allowed. Edit: There is a [check](https://github.com/elm-lang/Elm/blob/master/compiler/Transform/Canonicalize/Type.hs#L57) but it seems to be done lazily, when the type alias is used.

**Comment by Warry** [comment]
Seems to be fixed in 0.15:

```
Type Main.Y has too many arguments.

7| type alias Y2 = Y {}
Expecting 0, but got 1.
```

**Comment by mgold** [comment]
Confirmed, thanks.
